das-jdbc/src/main/java/org/wso2/das/jdbcdriver/common/SQLParser.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.das.jdbcdriver.common;
+
+import org.gibello.zql.ParseException;
+import org.gibello.zql.ZExpression;
+import org.gibello.zql.ZFromItem;
+import org.gibello.zql.ZQuery;
+import org.gibello.zql.ZSelectItem;
+import org.gibello.zql.ZStatement;
+import org.gibello.zql.ZqlParser;
+import org.wso2.das.jdbcdriver.aggregateFunctions.MaxAggrFunction;
+import org.wso2.das.jdbcdriver.aggregateFunctions.MinAggrFunciton;
+import org.wso2.das.jdbcdriver.aggregateFunctions.SumAggrFunction;
+import org.wso2.das.jdbcdriver.aggregateFunctions.CountAggrFunction;
+import org.wso2.das.jdbcdriver.expressions.AndExpression;
+import org.wso2.das.jdbcdriver.expressions.AsteriskExpression;
+import org.wso2.das.jdbcdriver.expressions.ColumnName;
+import org.wso2.das.jdbcdriver.expressions.Expression;
+import org.wso2.das.jdbcdriver.expressions.GeneralExpression;
+import org.wso2.das.jdbcdriver.expressions.ORExpression;
+import org.wso2.das.jdbcdriver.expressions.RelationOpExpression;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+
+/**
+ * This class have the functions required for parsing the SQL. ZQL Sql parser is used for this.
+ */
+public class SQLParser {
+
+    private List<Object[]> queryEnvironment = new ArrayList<Object[]>();
+    private Expression whereExpression;
+    private String sTableName;
+    private String sql;
+
+    public SQLParser(String sql) {
+        this.sql = sql;
+    }
+
+    /**
+     * Parse the SQL Query String by using the ZQL parser.
+     */
+    public void parse() throws SQLException {
+        System.out.println("SQL:" + sql);
+        if (sql.length() > 0) {
+            if (sql.charAt(sql.length() - 1) != ';') {
+                sql = sql.concat(";");
+            }
+        }
+        InputStream is = new ByteArrayInputStream(sql.getBytes());
```

Comment:
The try-with-resource statement could be used here so that closing this stream is handled automatically.